dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,372 @@
+            return 0
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1fees
+    def get_transaction_fee_estimate(self) -> int:
+        """Calculate the transaction fee estimate for a transaction given current fee rates
+        Returns:
+            The amount of estimated transaction fee cost for the network
+        """
+        _log.info("[BINANCE] Fetching the send fee...")
+        response = self._call_node_api("fees")
+        for fee_block in response:
+            if "fixed_fee_params" in fee_block:
+                if fee_block["fixed_fee_params"]["msg_type"] == "send":
+                    return fee_block["fixed_fee_params"]["fee"]  # fixed fee: 37500 (0.000375 BNB)
+        _log.info("[BINANCE] Fetch failed; resorting to saved value for fixed fee.")
+        return SEND_FEE  # SET GLOBALLY
+
+    # https://docs.binance.org/api-reference/node-rpc.html#6110-query-block
+    def get_current_block(self) -> int:
+        """Get the current latest block number of the network
+        Returns:
+            The latest known block number on the network
+        """
+        # if no "height" parameter is provided, latest block is fetched:
+        response = self._call_node_rpc("block", {})
+        current_block = response["result"]["block"]["header"]["height"]
+        return int(current_block)
+
+    def should_retry_broadcast(self, last_sent_block: int) -> bool:
+        """Check whether a new broadcast should be attempted, given a number of blocks past (for L5)
+        Args:
+            last_sent_block: The block when the transaction was last attempted to be sent
+        Returns:
+            Boolean whether a broadcast should be re-attempted
+        """
+        return self.get_current_block() - int(last_sent_block) > BLOCK_THRESHOLD
+
+    def get_network_string(self) -> str:
+        """Get the network string for this blockchain. This is what's included in l5 blocks or sent to matchmaking
+        Returns:
+            Network string
+        """
+        return f"binance {'testnet' if self.testnet else 'mainnet'}"
+
+    def get_private_key(self) -> str:
+        """Get the private key for this network
+        Returns:
+            string of the private key
+        """
+        return self.b64_private_key
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1accountaddress
+    def _fetch_account(self):
+        """Fetch the account metadata for an address
+        Returns:
+            response containing account metadata
+        """
+        _log.info(f"[BINANCE] Fetching address metadata for {self.address}")
+        path = f"account/{self.address}"  # params expected inside path string
+        try:
+            response = self._call_node_api(path)
+            return response
+        except exceptions.InterchainConnectionError:
+            _log.warning("[BINANCE] Non 200 response from Binance node. May have been a 500 Bad Request or a 404 Not Found.")
+            return 0
```

Comment:
Return 0? This is a completely different data type from the other return. What case would you ever want to get the integer 0 from `_fetch_account`? If you want to indicate an error, throw an exception here instead. Don't let this function have multiple return data types that have to be checked by the caller

Reply:
Copypasta mistake from similar error handling issue in `check_balance()`.  Removed.